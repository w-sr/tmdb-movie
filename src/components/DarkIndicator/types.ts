export interface Props {
  loading: boolean;
}
