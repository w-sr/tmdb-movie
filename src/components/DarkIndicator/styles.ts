import styled from 'styled-components/native';
import {Layout} from '../../lib';
import theme from '../../theme';

export const StyledWrapper = styled.View`
  position: absolute;
  align-items: center;
  justify-content: center;
  width: ${Layout.screen.width}px;
  height: ${Layout.screen.height}px;
`;

export const StyledBackground = styled.View`
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: ${theme.colors.black.hex};
  opacity: 0.5;
`;
