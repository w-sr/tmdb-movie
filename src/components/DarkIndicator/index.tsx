import React, {FC} from 'react';
import {ActivityIndicator} from 'react-native';
import theme from '../../theme';
import {StyledBackground, StyledWrapper} from './styles';
import {Props} from './types';

const DarkIndicator: FC<Props> = ({loading}) =>
  loading ? (
    <StyledWrapper>
      <StyledBackground />
      <ActivityIndicator color={theme.colors.white.hex} />
    </StyledWrapper>
  ) : (
    <></>
  );

export default DarkIndicator;
