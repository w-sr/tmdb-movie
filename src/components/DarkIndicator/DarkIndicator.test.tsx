import React from 'react';
import renderer from 'react-test-renderer';
import DarkIndicator from '.';

test('renders correctly', () => {
  const tree = renderer.create(<DarkIndicator loading={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
