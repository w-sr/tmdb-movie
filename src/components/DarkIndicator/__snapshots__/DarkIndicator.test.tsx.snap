// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 1334,
      "justifyContent": "center",
      "position": "absolute",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#000000",
        "bottom": 0,
        "left": 0,
        "opacity": 0.5,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <ActivityIndicator
    color="#FFFFFF"
  />
</View>
`;
